apps/src/componentLibrary/tooltip/Tooltip.tsx

```diff
@@ -60,31 +62,39 @@ export const TooltipOverlay: React.FunctionComponent<TooltipOverlayProps> = ({
  * Design System: Tooltip Component.
  * Renders Tooltip for a given html element (Needs to be wrapped in TooltipOverlay, see stories or README.md).
  */
-const Tooltip: React.FunctionComponent<TooltipProps> = ({
-  text,
-  tooltipId,
-  iconLeft,
-  iconRight,
-  direction = 'onTop',
-  className,
-  size = 'm',
-}) => {
-  return (
-    <div
-      id={tooltipId}
-      className={classnames(
-        moduleStyles.tooltip,
-        moduleStyles[`tooltip-${direction}`],
-        moduleStyles[`tooltip-${size}`],
-        className
-      )}
-      role="tooltip"
-    >
-      {iconLeft && <FontAwesomeV6Icon {...iconLeft} />}
-      <span className={moduleStyles.tooltipText}>{text}</span>
-      {iconRight && <FontAwesomeV6Icon {...iconRight} />}
-    </div>
-  );
-};
+const Tooltip = React.forwardRef<HTMLDivElement, TooltipProps>(
+  (
+    {
+      text,
+      tooltipId,
+      iconLeft,
+      iconRight,
+      direction = 'onTop',
+      className,
+      size = 'm',
+      style = {},
+    },
+    ref
+  ) => {
+    return (
+      <div
+        ref={ref}
```

Comment:
Unfamiliar with `forwardRef`...what does it mean to assign the ref to an HTML element attribute `ref`?

Reply:
Oh, `ref`s are awesome, though specific and not always needed feature of React! `ref` is used to access DOM element inside React code/component. Here it's used to get `Tooltip` DOM element height and width since we can't predict them in any other way.

`forwardRef` is a method of React that allows you to pass `ref` to the component as a prop. (Without forwardRef you'll be only able to set a ref to an html element inside your component, meaning you won't be able to access this dom element at current component's parent component and you won't be able to pass ref as a prop to this component)

You can read more about them here:
https://react.dev/learn/manipulating-the-dom-with-refs
https://react.dev/reference/react/forwardRef

Reply:
Oh cool I didn't realize you could assign a ref directly to a native HTML element, although that seems now fairly common now that I look at docs/other examples in our code. Thanks!